buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -383,20 +383,43 @@ class VagrantTestPlugin implements Plugin<Project> {
         assert project.tasks.packagingTest != null
         Task packagingTest = project.tasks.packagingTest
 
+        assert project.tasks.clean != null
+        Task clean = project.tasks.clean
+
         /*
          * We always use the main project.rootDir as Vagrant's current working directory (VAGRANT_CWD)
          * so that boxes are not duplicated for every Gradle project that use this VagrantTestPlugin.
          */
+        File vagrantFile = new File(project.rootDir, 'Vagrantfile')
         def vagrantEnvVars = [
                 'VAGRANT_CWD'           : "${project.rootDir.absolutePath}",
-                'VAGRANT_VAGRANTFILE'   : 'Vagrantfile',
+                'VAGRANT_VAGRANTFILE'   : vagrantFile.name,
                 'VAGRANT_PROJECT_DIR'   : "${project.projectDir.absolutePath}"
         ]
 
+        File vagrantMarkersDir = new File(project.rootProject.buildDir, "vagrant")
+        clean.delete vagrantMarkersDir
```

Comment:
I see, it is because you are doing this in the root project, I did not notice that before. I think this is going to be problematic long term. We either need to have shared tasks that are setup on the root project for these boxes, or make them per project. I think the latter is better. It would mean passing the working dir/vagrantfile to each vagrant command, and so the same box name, but in different gradle projects, would be different vm instances.